crates/sui-bridge-indexer/src/main.rs

```diff
@@ -155,9 +155,9 @@ async fn main() -> Result<()> {
         );
         indexer.start().await?;
     }
-    // We are not waiting for the sui tasks to finish here, which is ok.
-    futures::future::join_all(vec![subscription_indexer_fut, sync_indexer_fut]).await;
-
+    // These tasks should not finish
+    subscription_indexer_fut.await.unwrap().unwrap();
+    sync_indexer_fut.await.unwrap().unwrap();
```

Comment:
why not use join?

Reply:
* `join_all` won't return until all tasks are finished. So if one task errors, we won't immediately know
* `try_join!` gives a vector of `Result<>` which i don't bother to unwrap

this way feels the easiest and less error-prone

Reply:
> join_all won't return until all tasks are finished. So if one task errors, we won't immediately know


if you rmr it took us a while to figure out why there is no Eth backfill tasks, this is why